pkg/controller/storagecluster/storagecluster.go

```diff
@@ -631,14 +631,19 @@ func (c *Controller) syncStorageCluster(
 	}
 
 	pxVer30, _ := version.NewVersion("3.0")
-	if pxutil.GetPortworxVersion(cluster).GreaterThanOrEqual(pxVer30) { // Preflight should only run on PX version 3.0.0 and above
+	if pxutil.IsFreshInstall(cluster) && pxutil.GetPortworxVersion(cluster).GreaterThanOrEqual(pxVer30) { // Preflight should only run on PX version 3.0.0 and above
 		// If preflight failed, or previous check failed, reconcile would stop here until issues got resolved
 		if err := c.runPreflightCheck(cluster); err != nil {
 			if updateErr := c.updateStorageClusterState(cluster, corev1.ClusterStateDegraded); updateErr != nil {
 				logrus.Errorf("failed to update StorageCluster status. %v", updateErr)
 			}
 			return fmt.Errorf("preflight check failed for StorageCluster %v/%v: %v", cluster.Namespace, cluster.Name, err)
 		}
+	} else {
+		if cluster.Annotations == nil {
+			cluster.Annotations = make(map[string]string)
+		}
+		cluster.Annotations[pxutil.AnnotationPreflightCheck] = "false"
```

Comment:
This should change to - 
```
		if _, exists := cluster.Annotations[pxutil.AnnotationPreflightCheck]; !exists {
                         if cluster.Annotations == nil {
              			cluster.Annotations = make(map[string]string)
                         }
                         cluster.Annotations[pxutil.AnnotationPreflightCheck] = "false"
		}
		

```

Reply:
We don't want to set the value if one already exists. Basically, if the user or the operator has already changed the value, we don't want to change it again.

Reply:
Actually yeah I did another commit after that .. which changed it like this `if cluster.Annotations == nil {
			cluster.Annotations = make(map[string]string)
		}
		// Only set to false if its not set or empty
		if _, ok := cluster.Annotations[pxutil.AnnotationPreflightCheck]; !ok {
			cluster.Annotations[pxutil.AnnotationPreflightCheck] = "false"
		} else if cluster.Annotations[pxutil.AnnotationPreflightCheck] == "" {
			cluster.Annotations[pxutil.AnnotationPreflightCheck] = "false"
`